src/gzip.rs

```diff
@@ -395,18 +391,19 @@ impl Header {
                 GZIP_ID
             ));
         }
-        let compression_method = reader.read_u8()?;
+        let compression_method = buf[2];
         if compression_method != COMPRESSION_METHOD_DEFLATE {
             return Err(invalid_data_error!(
                 "Compression methods other than DEFLATE(8) are \
                  unsupported: method={}",
                 compression_method
             ));
         }
-        let flags = reader.read_u8()?;
-        this.modification_time = reader.read_u32::<LittleEndian>()?;
-        this.compression_level = CompressionLevel::from_u8(reader.read_u8()?);
-        this.os = Os::from_u8(reader.read_u8()?);
+        let flags = buf[3];
+        this.modification_time =
+            u32::from_le_bytes(TryInto::<[u8; 4]>::try_into(&buf[4..4 + 4]).unwrap());
```

Comment:
I prefer the following code because it can eliminate the use of `unwrap` (and `TryInto`):
```suggestion
            u32::from_le_bytes([buf[4], buf[5], buf[6], buf[7]]);
```